File: solutions/leetcode/largest-3-same-digit-number-in-string/main.cpp
class Solution {
   public:
    string largestGoodInteger(string num) {
        string s;
        string w = num.substr(0, 2);
        for (int i = 2; i < num.size(); ++i) {
            w += num[i];
            if (w[0] == w[1] && w[1] == w[2]) {
                s = max(s, w);
            }
            w = w.substr(1, 2);
        }
        return s;
    }
};

class Solution {
   public:
    string largestGoodInteger(string num) {
        string result = "";
        for (int l = 0; l < num.size() - 2; ++l) {
            if (num[l] == num[l + 1] && num[l] == num[l + 2] &&
                (result == "" || num[l] > result[0])) {
                result = num.substr(l, 3);
            }
        }
        return result;
    }
};

File: solutions/leetcode/largest-3-same-digit-number-in-string/main.py
class Solution:

    def largestGoodInteger(self, num: str) -> str:
        result = ""
        for r in range(3, len(num) + 1):
            l = r - 3
            if num[l] == num[l + 1] == num[l + 2] and (result == ""
                                                       or num[l] > result[0]):
                result = num[l:r]
        return result
